### ISSUE TITLE
Update meow version

### ISSUE THREAD

**Issue by scott-8** [fi3ework/vite-plugin-checker#255]
### Is your feature request related to a problem? Please describe.

The latest versions of `stylelint` use `meow@^10.1.5`. `vite-plugin-checker` uses `meow@^10.1.5`.

### Describe the solution you'd like

It would be nice if this package could use the version of `meow` that is already installed by `stylelint`.

### Validations

- [X] Read the [docs](https://github.com/fi3ework/vite-plugin-checker#readme).
- [X] Read the [Contributing Guidelines](https://github.com/fi3ework/vite-plugin-checker/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that asks for the same feature to avoid creating a duplicate.

**Comment by fi3ework** [comment]
meow is treated as an optional peer dependency of vite-plugin-checker. As the document says:

> (Optional but highly recommended) Install meow@^9.0.0 with your package manager. It's needed because of Stylelint dependents on it. It's probably working fine even it's not installed as it's accessed as a phantom dependency. But when you set hoist=false of pnpm. It won't be accessible anymore without explicit installation.

So you could not install meow if it's resolved for vite-plugin-checker.

**Comment by scott-8** [comment]
I should have been more specific, everything resolves OK. The issue is that meow v9 will install an older version of `semver` which triggers an alert for [this CVE](https://github.com/advisories/GHSA-c2qf-rxjj-qqgw) in code scans.

Meow versions starting at 10 don't have this problem/use newer dependencies.

**Comment by fi3ework** [comment]
Suppose stylelint relies on v9 meow. v9 meow will be installed regardless vite-plugin-checker is installed.

**Comment by Magiczne** [comment]
@fi3ework not really. Tried to update to stylelint v15 with vite-plugin-checker 0.6.1, and only installed version of meow is 10, as it is only a peer dependency. 

The only way to fix this is to explicitly declare `"meow": "^9.0.0"` in the package.json as dependency. 

Please allow usage of meow ^10 to be more compatibile with stylelint, as currently including dependency in the application-level package.json just for another lib to work is not great.

**Comment by fi3ework** [comment]
@Magiczne I haven't used yarn3, but I guess it behaves like pnpm with `hoist=false` which is suggested to install meow as a dep explicitly.
I admit installing `meow` by user is not friendly for the user. I'm working on next breaking version that will split the plugin into core and checkers. After that, the usage would be like:

package.json
```json
"vite-plugin-checker": "some version", 
"@vite-plugin-checker/stylelint": "some version",
"styelint": "some version"
```

vite config
```js
import { checker } from 'vite-plugin-checker'
import { checker as stylelint } from '@vite-plugin-checker/stylelint'

export default {
  plugins: [ checker([stylelint()]) ]
}
```

In this way, user will not touch the details and lower dependencies. 

Nevertheless, it still will be hard to sync meow with stylelint. We can bump meow to keep in sync with stylelint. But it still will lead backport fix for older version which is hard to maintain.

**Comment by Magiczne** [comment]
Thanks for explnation! I think that if there will be separate packages it will be definitely easier. For now as workaround works we'll just stick to it and wait for next version

**Comment by fi3ework** [comment]
And the dependencies count of `vite-plugin-checker@0.7` + `@vite-plugin-checker/stylelint` will be less that `vite-plugin-checker@0.6` as it will only carry the dependencies that will be used by each checker package. In the old version, some of them are installed directly in vite-plugin-checker for convenient.

**Comment by tryforceful** [comment]
I get this error when trying to use `meow@10.1.5` with the current latest version of `vite-plugin-checker@0.6.1`:

```
error during build:
Error [ERR_REQUIRE_ESM]: require() of ES Module <mydir>/node_modules/.pnpm/meow@10.1.5/node_modules/meow/index.js from <mydir>/node_modules/.pnpm/vite-plugin-checker@0.6.1_eslint@8.44.0_meow@10.1.5_stylelint@15.10.1_typescript@5.1.6_vite@4.3.9_vue-tsc@1.8.3/node_modules/vite-plugin-checker/dist/cjs/checkers/stylelint/options.js not supported.
Instead change the require of index.js in <mydir>/node_modules/.pnpm/vite-plugin-checker@0.6.1_eslint@8.44.0_meow@10.1.5_stylelint@15.10.1_typescript@5.1.6_vite@4.3.9_vue-tsc@1.8.3/node_modules/vite-plugin-checker/dist/cjs/checkers/stylelint/options.js to a dynamic import() which is available in all CommonJS modules.
```

Using `meow@9` didn't generate this error. Is this related to this issue?

**Comment by fi3ework** [comment]
@tryforceful Not the same issue. Please file an new issue with a minimal reproduction please.
